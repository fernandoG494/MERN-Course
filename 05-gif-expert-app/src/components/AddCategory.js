import React, {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ setCategories }) => {
    const [inputText, setInputText] = useState('');
    
    const handleInputChange = (e) => {
        setInputText(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputText.trim().length > 2){
            setCategories(cats => [inputText, ...cats]);
            setInputText('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputText}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
