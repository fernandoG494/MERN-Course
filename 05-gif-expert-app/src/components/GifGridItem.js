import React from 'react'

function GifGridItem({title, url}) {
    // console.log(title, url);
    
    return (
        <div>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

export default GifGridItem
