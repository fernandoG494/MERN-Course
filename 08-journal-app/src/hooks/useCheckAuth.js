import { useEffect } from 'react';
import { FirebaseAuth } from '../firebase/config';
import { onAuthStateChanged } from 'firebase/auth';
import { useSelector, useDispatch } from 'react-redux';
import { login, logout } from '../store/auth/authSlice';

const useCheckAuth = () => {
    const dispatch = useDispatch();
    const { status } = useSelector(state => state.auth);

    useEffect(() => {
        onAuthStateChanged(FirebaseAuth, async( user ) => {
            if( !user ) return dispatch(logout());
            const { uid, email, displayName, photoURL } = user;

            dispatch(login({ uid, email, displayName, photoURL }));
        });
    }, []);

    return { status };
};

export default useCheckAuth;
