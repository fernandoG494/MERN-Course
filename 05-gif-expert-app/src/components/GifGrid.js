import React, { useState, useEffect } from 'react';
import GifGridItem from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

const GifGrid = ({category}) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs(category).then(imgs => setImages(imgs));
    }, [category]);     // vacio se ejecuta una vez

    return (
        <>
            <h3> { category } </h3>
            <div className='card-grid'>
                <ol>
                    {   
                        images.map(img => (
                            <GifGridItem
                                key={img.title}
                                {...img}
                            />
                        ))
                    }
                </ol>
            </div>
        </>
    )
}

export default GifGrid
