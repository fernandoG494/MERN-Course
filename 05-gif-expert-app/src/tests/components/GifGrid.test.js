import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");


describe('tests in GifGrid', () => {
    const category = 'death';

    test('should match snapshot', () => {
        const gifs = {
            id: 'ABC',
            url: 'https://localhost/something.jpg',
            title: 'title'
        };
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when useFetchGifs', () => {
        const gifs = {
            id: 'ABC',
            url: 'https://localhost/something.jpg',
            title: 'title'
        };
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
    });
});