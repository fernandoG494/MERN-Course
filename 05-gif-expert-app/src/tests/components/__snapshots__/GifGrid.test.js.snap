// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifGrid should match snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    death
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`tests in GifGrid should show items when useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    death
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
